frontend/src/routes/accounts/ProfilePhoto.tsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import loader from '../../loader.gif';
+import placeholder from '../../placeholder.jpg';
+import './ProfilePhoto.css'
+
+interface Props {
+    username?: string
+};
+
+const ProfilePhoto: React.FC<Props> = ({ username }) => {
+    type ACTIONTYPE =
+        | { type: "fetch_init"; }
+        | { type: "fetch_success"; imageUrl: string; canUpload: boolean; }
+        | { type: "fetch_fail"; };
+
+    const initialState = {
+        imageUrl: loader,
+        tooltipText: 'Loading...',
+        isLoading: true,
+        isError: false,
+        canUpload: false,
+        newImageUrl: ''
+    };
+
+    const profilePhotoReducer = (state: typeof initialState, action: ACTIONTYPE) => {
```

Comment:
Ovo je moglo lepše koristeći destruktuiranje objekata:
```javascript
const profilePhotoReducer = (state: typeof initialState, action: ACTIONTYPE) => {
    const {type, imageUrl, canUpload} = action;
    switch (type) {
        case 'fetch_init':
            return {
                ...state,
                isLoading: true,
                isError: false
            };
        case 'fetch_success':
            return {
                ...state,
                isLoading: false,
                isError: false,
                imageUrl,
                canUpload,
                tooltipText: canUpload ?
                    'Click here to upload a new profile photo' :
                    `${username}'s profile photo`
            };
        case 'fetch_fail':
            return {
                ...state,
                isError: true,
                imageUrl: 'error.png',
                tooltipText: 'Error loading profile photo'
            };
        default:
            throw new Error();
    }
};
 ```

Reply:
Još bitnije od toga jeste što, koliko vidim, reducer unutar komponente uopšte nije dobra praksa i možda čak i ne radi. Ukoliko se promeni korisničko ime, možda može da se desi da korisničko ime unutar tooltip ostane isto. Najbolje je pri `fetch_success` dostaviti reducer-u korisničko ime kroz `dispatch`.